drivers/mtd/smart.c

```diff
@@ -4560,7 +4561,7 @@ static int smart_writesector(FAR struct smart_struct_s *dev,
       /* Subtract dev->minwearlevel from all wear levels */
 
       offset = dev->minwearlevel;
-      finfo("Reducing wear level bits by %d\n", offset);
+      finfo("Reducing wear level bits by %zd\n", offset);
```

Comment:
```suggestion
      finfo("Reducing wear level bits by %zu\n", offset);
```
`offset` here is defined as unsigned size type.